Adding openms

asmaahassan90 opened the pull request:
OpenMS formula is working just for ubuntu for now, till we resolve qt issues for centos.

This merge request has to be approved first https://gitlab.kaust.edu.sa/kaust-rc/docker-registry/merge_requests/5, before proceeding with approving this one.

asmaahassan90 committed:
Adding current version of OpenMS formula. Troubleshooting in progress...

asmaahassan90 committed:
fix paths

Commit:
fixing build - contrib packages

Commit:
fixing build

Commit:
build problems fixed, but still having issues related to xserver

Commit:
Successfull installation! had to disable GUI - test needs refining

Commit:
success on ubuntu 16 - needs qt fix for cento6

Commit:
Fix qt path problem for centos6

Commit:
adding the KAUST_SKIP tag to test only on ubuntu

antoniokaust commented on openms.rb:
```diff
@@ -0,0 +1,45 @@
+  sha256 "61e7d24890abb6f462e2f026efb0dc88b84639b228d96e919c8074d3b11d286e"
+
+## Adding the images to be skipped while testing 
+# KAUST_SKIP centos:6 centos:7
```
@asmaahassan90 Do you think we should also check for these in the `install ` section? Otherwise ho do we prevent users from installing it on CentOS? We could probably print a message saying only supported on Ubuntu.

asmaahassan90 replied:
I was thinking of adding a comment in the readme file, stating the requirements as it needs to have qt4 installed and mentioning that it only works for ubuntu for now till we fix this thing with centos.

antoniokaust replied:
We have a README for this? OR do you mean on our front page for the repo?

asmaahassan90 replied:
I meant our front page. Also I believe we might need to do this for some other future formulas, so we gotta have some kind of standard for it, whether we agree to put it on out front page or add a readme file for each formula that needs certain requirements or works for certain OS.

antoniokaust replied:
I guess README is good for now. We can always change later on. So please include it in this patch.

Nahlaelmaraashly25 commented on openms.rb:
```diff
@@ -0,0 +1,45 @@
+  depends_on "eigen"
+
+  def install
+    ENV.prepend_create_path "PATH", "/usr/lib64/qt4/bin" if File.exist?("/etc/redhat-release")
```
@asmaahassan90  do we need the part containing redhat-release for Ubuntu or just CentOS ?

asmaahassan90 replied:
it was added to check if OS is centos or redhat to set qt path properly. but we might not need it right now. Will edit the formula and set it to default

asmaahassan90 committed:
update formula

remove check for centos OS and the part of setting qt path since Ubuntu has no issues with detecting qt

asmaahassan90 committed:
updating homepage url to use https

asmaahassan90 committed:
adding  "know issues" section

adding notes about openms formula

asmaahassan90 committed:
fixing identation

antoniokaust commented:
Di local tests pass?

asmaahassan90 commented:
mine did, on both ubuntu trusty and xenial
